data_prep/generate_climos.py

```diff
@@ -189,77 +159,117 @@ def update_climo_time_meta(fp, file_type=Cmip5File):
       - PCIC CMIP5 style file path
     '''
     
-    cf = file_type(fp)
-    nc = Dataset(fp, 'r+')
-    timevar = nc.variables['time']
+    cf = CFDataset(filepath)
+    time_var = cf.variables['time']
 
     # Generate new time/climo_bounds data
-    if cf.freq == 'yrClim':
+    if cf.time_resolution == 'yearly':
         time_types = ('annual')
     else:
         time_types = ('monthly', 'seasonal', 'annual')
 
-    times, climo_bounds = generate_climo_time_var(ss2d(cf.t_start), ss2d(cf.t_end), time_types)
+    start_date, end_date = cf.time_range
+    # Migration note: start_date, end_date may need to be converted to real datetime.datetime values
+    times, climo_bounds = generate_climo_time_var(start_date, end_date, time_types)
 
-    timevar[:] = date2num(times, timevar.units, timevar.calendar)
+    time_var[:] = date2num(times, time_var.units, time_var.calendar)
 
     # Create new climatology_bounds variable and required bnds dimension
-    timevar.climatology = 'climatology_bounds'
-    nc.createDimension('bnds', 2)
-    climo_bnds_var = nc.createVariable('climatology_bounds', 'f4', ('time', 'bnds', ))
-    climo_bnds_var.calendar = timevar.calendar
-    climo_bnds_var.units = timevar.units
-    climo_bnds_var[:] = date2num(climo_bounds, timevar.units, timevar.calendar)
-
-    nc.close()
+    time_var.climatology = 'climatology_bounds'
+    cf.createDimension('bnds', 2)
+    climo_bnds_var = cf.createVariable('climatology_bounds', 'f4', ('time', 'bnds', ))
+    climo_bnds_var.calendar = time_var.calendar
+    climo_bnds_var.units = time_var.units
+    climo_bnds_var[:] = date2num(climo_bounds, time_var.units, time_var.calendar)
 
+    cf.close()
 
-def main(args):
-    vars_ = '|'.join(args.variables)
-    test_files = iter_matching(args.basedir, re.compile('.*({}).*(_rcp26|_rcp45|_rcp85|_historical_).*r1i1p1.*nc'.format(vars_)))
 
-    if args.dry_run:
-        for f in test_files:
-            print(f)
-        sys.exit(0)
+def climo_output_filepath(output_dir, input_file, t_start, t_end):
+    '''Join the output directory to the output filename for this file'''
+    return os.path.realpath(os.path.join(output_dir, input_file.climo_output_filename(t_start, t_end)))
 
-    FileType = ClimdexFile if args.climdex else Cmip5File
 
-    for fp in test_files:
-        log.info(fp)
+def main(args):
+    if args.basedir:
+        variables = '|'.join(args.variables)
+        filepaths = iter_matching(
+            args.basedir,
+            re.compile('.*({}).*_(historical)?((?<=l)\+(?=r))?(rcp26|rcp45|rcp85)?_.*r\di\dp\d.*nc'.format(variables))
```

Comment:
I'm good with that. I do find the capacity additionally to read the set of input files from a file is useful, especially given the extremely long filepaths when testing from a dev machine.